backend/src/apiserver/storage/experiment_store.go

```diff
@@ -26,33 +26,69 @@ type ExperimentStore struct {
 	resourceReferenceStore *ResourceReferenceStore
 }
 
-func (s *ExperimentStore) ListExperiments(opts *list.Options) ([]*model.Experiment, string, error) {
-	errorF := func(err error) ([]*model.Experiment, string, error) {
-		return nil, "", util.NewInternalServerError(err, "Failed to list experiments: %v", err)
+// Runs two SQL queries in a transaction to return a list of matching experiments, as well as their
+// count. The count does not reflect the page size.
+func (s *ExperimentStore) ListExperiments(opts *list.Options) ([]*model.Experiment, int32, string, error) {
+	errorF := func(err error) ([]*model.Experiment, int32, string, error) {
+		return nil, 0, "", util.NewInternalServerError(err, "Failed to list experiments: %v", err)
 	}
 
-	sql, args, err := opts.AddToSelect(sq.Select("*").From("experiments")).ToSql()
+	sqlBuilder := opts.AddFilterToSelect(sq.Select("*").From("experiments"))
+
+	// SQL for row count
+	countSql, countArgs, err := sq.Select("count(*)").FromSelect(sqlBuilder, "rows").ToSql()
```

Comment:
Why is this better? Is it to guarantee the query optimizer doesn't actually read all rows then return their count?
The idea is to wrap a count query around the filter query without changing anything in it, so that if the base query changes, this one doesn't have to.

Reply:
you could instead make what's being selected as a parameter to achieve the same goal but with more concise sql query.

Reply:
something like 
opts.AddFilterToSelect(sq.Select("*").From("experiments"))
opts.AddFilterToSelect(sq.Select(count("*")).From("experiments"))

Reply:
Done, using simpler queries counting filtered rows in the original table directly.